### PR TITLE
Re-add backwards compatibility for URLs using old UserSettings plugin.

### LINKED ISSUE #8923: Some static Image graphs are broken
See at: http://developer.piwik.org/api-reference/reporting-api-metadata#static-image-graphs

Some of the graphs are broken with error message "Invalid API module and/or action"

This was reported in forums: http://forum.piwik.org/read.php?9,128022

This regressed during 2.15.0 beta cycle. 

Steps 
 * Check why our UI tests for static image graph didn't catch this bug
 * Fix issue

### PR DESCRIPTION
**Pull request by diosmosis**
As title. Includes system tests so BC break will be detected in the future. And includes tweaks to UI tests for more coverage.

Fixes #8923

### CODE REVIEW

File: core/testMinimumPhpVersion.php [hunk 1]
```diff
@@ -78,7 +78,7 @@
      * @return bool
      */
     function Piwik_ShouldPrintBackTraceWithMessage()
-    {
+    {return true;
```
Comment (mattab) [inline core/testMinimumPhpVersion.php]: typo

File: plugins/DevicesDetection/DevicesDetection.php [hunk 2]
```diff
@@ -49,4 +50,33 @@ public function extendVisitorDetails(&$visitor, $details)
         $visitor['browserVersion']           = $instance->getBrowserVersion();
     }
 
+    public function renameUserSettingsModuleAndAction(&$module, &$action)
```
Comment (tsteur) [inline plugins/DevicesDetection/DevicesDetection.php]: Can we remove it again in Piwik 3.0 maybe? We could add a reminder to remove it in `DeprecatedMethodsTest`

File: plugins/DevicesDetection/DevicesDetection.php [hunk 3]
```diff
@@ -49,4 +50,33 @@ public function extendVisitorDetails(&$visitor, $details)
...
+            'getBrowserType' => 'getBrowserEngines',
+        );
+
+        if ($module == 'UserSettings' && array_key_exists($action, $movedMethods)) {
+            $module = 'DevicesDetection';
+            $action = $movedMethods[$action];
+        }
+
+        if ($module == 'UserSettings' && ($action == 'getResolution' || $action == 'getConfiguration')) {
+            $module = 'Resolution';
```
Comment (tsteur) [inline plugins/DevicesDetection/DevicesDetection.php]: Should this one be in the `Resolution` plugin and the ones below in the other plugin? The plugins might not be enabled and moving them to the actual plugins will avoid having to check whether they are enabled here. 

Of course if eg `DevicePlugins` is disabled the BC for `getPlugin` doesn't work but in this case we should rather respect the users choice to have the plugin disabled

File: plugins/DevicesDetection/DevicesDetection.php (hunk 2 above)
Comment (diosmosis) [inline plugins/DevicesDetection/DevicesDetection.php]: Added to DeprecatedMethodsTest.

**Comment by diosmosis** [comment]
Ready for another review/merge.

**Pull request merged**